src/main/java/bob/Parser.java

```diff
@@ -0,0 +1,182 @@
+package bob;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import bob.command.AddDeadlineCommand;
+import bob.command.AddEventCommand;
+import bob.command.AddTodoCommand;
+import bob.command.Command;
+import bob.command.DeleteCommand;
+import bob.command.ExitCommand;
+import bob.command.ListCommand;
+import bob.command.ListDueInCommand;
+import bob.command.ListOnDateCommand;
+import bob.command.MarkCommand;
+import bob.exception.BobException;
+import bob.exception.EmptyDescriptionException;
+import bob.exception.InvalidCommandException;
+import bob.exception.InvalidDateTimeException;
+import bob.exception.InvalidDaysException;
+import bob.exception.InvalidTaskIndexException;
+import bob.exception.ParameterNotFoundException;
+
+public class Parser {
+    private static final String EXIT = "exit";
+    private static final String LIST = "list";
+    private static final String MARK = "mark";
+    private static final String UNMARK = "unmark";
+    private static final String TODO = "todo";
+    private static final String DEADLINE = "deadline";
+    private static final String EVENT = "event";
+    private static final String DELETE = "delete";
+
+    private static final String DATE_PATTERN = "d/M/yyyy";
+    private static final DateTimeFormatter DATE_FORMATTER
+            = DateTimeFormatter.ofPattern(DATE_PATTERN);
+
+    private static final String DATETIME_PATTERN = DATE_PATTERN + " HHmm";
+    private static final DateTimeFormatter DATETIME_FORMATTER
+            = DateTimeFormatter.ofPattern(DATETIME_PATTERN);
+
+    private static String[] extractParameters(String parametersString,
+                                             String[] parameters) throws ParameterNotFoundException {
+        // TODO: generalise the method to any given parametersString, desiredParameters and separator
+        // TODO: perform ParameterNotFoundException checks correctly
```

Comment:
should remove comments before making the pull request

Reply:
sflr, I do think that these comments are helpful to communicate the loose ends to tie before the PR can be merged, or possible improvements after the PR is merged, which is why I left it in the PR.